pkg/operator/ceph/csi/spec.go

```diff
@@ -318,6 +319,11 @@ func (r *ReconcileCSI) startDrivers(ver *version.Info, ownerInfo *k8sutil.OwnerI
 		tp.EnableCSIAddonsSideCar = true
 	}
 
+	tp.EnableCSIEncryption = false
+	if strings.EqualFold(k8sutil.GetValue(r.opConfig.Parameters, "CSI_ENABLE_ENCRYPTION", "false"), "true") {
```

Comment:
Instead of exposing this setting to mount the configmap, I wonder if we can avoid it with one of these options:
1. Only add the encryption settings in the storage class, instead of having the configmap in the middle as suggested above
2. Or if we really need the configmap, the operator could always ensure the configmap exists (create an empty one if it doesn't exist yet), and always mount the configmap.

Reply:
> Instead of exposing this setting to mount the configmap, I wonder if we can avoid it with one of these options:
> 
> 1. Only add the encryption settings in the storage class, instead of having the configmap in the middle as suggested above

answered regarding the difficulty doing this [here](https://github.com/rook/rook/pull/9940#discussion_r842362624)

> 2. Or if we really need the configmap, the operator could always ensure the configmap exists (create an empty one if it doesn't exist yet), and always mount the configmap.

I think users should be managing the required configmap external to rook.
- They may want to preserve kms settings even when destroying the cluster.
- If Rook creates a configmap, it should also take care of cleanup. 

The above two points go against each other.